src/options/README

```diff
@@ -0,0 +1,225 @@
+      links      = []
+      read_only  = false
+      help       = "force output language (default is \"auto\"; see --output-lang help)"
+
+
+  If an alternate option is generated the linked options defined via attribute
+  links are not considered. If you want to define links for an alternate option
+  --no-<long> for an existing option <long>, you can define an alias with long
+  option no-<long>. This overwrites the default --no-<long> behaviour and
+  creates the linked options.
+
+
+Aliases
+=======
+
+  Aliases can be defined with the [[alias]] tag, which creates a new long
+  option and binds it to the list of long options specified via the 'links'
+  attributes.
+
+
+  The required attributes are:
+
+    category ... (string) common | expert | regular | undocumented
+    long     ... (string) long option name
+    links    ... (list)   list of long options to set
+
+  Optional attributes are:
+
+    help     ... (string) documentation (only option for category undocumented)
+
+
+  Example:
+
+    [[alias]]
+      category   = "regular"
+      long       = "smtlib-strict"
+      links      = ["--lang=smt2", "--output-lang=smt2", "--strict-parsing", "--default-expr-depth=-1", "--print-success", "--incremental", "--abstract-values"]
+      help       = "SMT-LIBv2 compliance mode (implies other options)"
+
+
+  This example creates a regular option with the long option name
+  'smtlib-strict', which links to the long options given as a list 'links'.
+  Calling
+
+    --smtlib-strict
+
+  is equivalent to specifying the options
+
+    --lang=smt2 --output-lang=smt2 --strict-parsing --default-expr-depth=-1 --print-success --incremental --abstract-values
+
+
+  It's also possible to pass an argument through to another option.
+
+  Example:
+
+    [[alias]]
+      category   = "undocumented"
+      long       = "output-language=L"
+      links      = ["--output-lang=L"]
+
+  This alias makes --output-language synonymous with --output-lang and passes
+  argument L through to --output-lang. The placeholer name (in this example
+  'L') of the argument can be arbitrarily chosen and can be referenced multiple
+  times in 'links'.
+
+
+
+Further information (the old option package)
+============================================
+
+  The options/ package supports a wide range of operations for responding to
+  an option being set. Roughly the three major concepts are:
+
+   - handler is to parse an option before setting its value.
+   - predicates to reject bad values for the option.
+   - notifies for dynamic dispatch after an option is assigned.
+
+  More details on each class of custom handlers.
+
+   - handler = ""
+
+    Handlers provide support for parsing custom types and parsing for options.
+    The signature for a handler call is:
+
+       T custom-option-handler(std::string option, std::string optarg,
+                               OptionsHandler* handler);
+
+    where T is the type of the option. The suggested implementation is to
```

Comment:
`implementation` -> `procedure` ?

Reply:
We are talking about specific implementations of handlers here, so this should be ok.